var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -160,6 +160,14 @@ def install(self, spec, prefix):
                 'ON' if '+hypre' in spec else 'OFF')
         ])
 
+        if self.compiler.name == 'intel' and str(self.version) == '12.6.2':
```

Comment:
since you don't know which version of compiler it applies to, maybe specify the one you tested by  `spec.satisfies('%intel@x.y.z')`. 

Also version comparison is not needed here, just write `spec.satisfies('@12.6.2')`.